ipaclient/install/client.py

```diff
@@ -2915,16 +2921,17 @@ def _install(options):
             # Particulary, SSSD might take longer than 6-8 seconds.
             while n < 10 and not found:
                 try:
-                    ipautil.run(["getent", "passwd", "admin@%s" % cli_domain])
+                    ipautil.run(["getent", "passwd", "%s@%s"
+                                % (principal, cli_domain)])
```

Comment:
The man page for ipa-client-install specifies that -p is used to provide a kerberos principal. This means that options.principal can have the format username or username@DOMAIN.COM.
If the latter format is provided, the code will call getent passwd username@DOMAIN.COM@domain.com and fail.